File: 693.交替位二进制数.cpp
/*
 * @lc app=leetcode.cn id=693 lang=cpp
 *
 * [693] 交替位二进制数
 */

// @lc code=start
class Solution
{
public:
    bool hasAlternatingBits(int n)
    {
        long num = n ^ (n >> 1);
        return (num & (num + 1)) == 0;
    }
};
// @lc code=end

File: 693.交替位二进制数.py
#
# @lc app=leetcode.cn id=693 lang=python3
#
# [693] 交替位二进制数
#

# @lc code=start
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        t = n % 2
        n //= 2
        while n:
            if t == n % 2:
                return False
            t = n % 2
            n //= 2
        return True
# @lc code=end

